snapshotter/demux/internal/mock/snapshotter.go

```diff
@@ -0,0 +1,205 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package mock
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/mount"
+	"github.com/containerd/containerd/snapshots"
+)
+
+// SnapshotterOpt defines behavior options for mock snapshotter objects.
+type SnapshotterOpt func(*Snapshotter)
+
+// Snapshotter is a mock whose behavior can be configured via options.
+type Snapshotter struct {
```

Comment:
How about just exposing all variables instead of having WithXxx methods?

Reply:
Haha yeah that's a better simplification.